test/Microsoft.ML.Functional.Tests/Prediction.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.SamplesUtils;
+using Xunit;
+
+namespace Microsoft.ML.Functional.Tests
+{
+    public partial class PredictionScenarios
+    {
+        /// <summary>
+        /// Reconfigurable predictions: The following should be possible: A user trains a binary classifier,
+        /// and through the test evaluator gets a PR curve, the based on the PR curve picks a new threshold
+        /// and configures the scorer (or more precisely instantiates a new scorer over the same predictor)
+        /// with some threshold derived from that.
+        /// </summary>
+        [Fact(Skip = "Blocked by issue #2465")]
+        public void ReconfigurablePrediction()
+        {
+            var mlContext = new MLContext(seed: 789);
+
+            // Get the dataset, create a train and test
+            var dataset = DatasetUtils.LoadHousingRegressionDataset(mlContext);
+            (var train, var test) = mlContext.BinaryClassification.TrainTestSplit(dataset, testFraction: 0.2);
+
+            // Create a pipeline to train on the housing data
+            var pipeline = mlContext.Transforms.Concatenate("Features", new string[] {
+                    "CrimesPerCapita", "PercentResidental", "PercentNonRetail", "CharlesRiver", "NitricOxides", "RoomsPerDwelling",
+                    "PercentPre40s", "EmploymentDistance", "HighwayDistance", "TaxRate", "TeacherRatio"})
+                .Append(mlContext.Transforms.CopyColumns("Label", "MedianHomeValue"))
+                .Append(mlContext.Regression.Trainers.OrdinaryLeastSquares());
+
+            var model = pipeline.Fit(train);
+
+            var scoredTest = model.Transform(test);
+            var metrics = mlContext.Regression.Evaluate(scoredTest);
```

Comment:
Should we be asserting the metrics are in a certain range?

Reply:
Good call on checking valid ranges. I added a Common library to add those sorts of checks to.

---
In reply to: [255167478](https://github.com/dotnet/machinelearning/pull/2470#discussion_r255167478) [](ancestors = 255167478)

Reply:
I think you missed checking that new function in.

---
In reply to: [255228907](https://github.com/dotnet/machinelearning/pull/2470#discussion_r255228907) [](ancestors = 255228907,255167478)